app/installer.sh

```diff
@@ -18,210 +18,261 @@
-  exit 1;
-fi
-if ! type "gzip" > /dev/null 2>&1; then
-  printf "ERROR\tgzip cannot be found\n"
-  exit 1;
-fi
+main() {
+
+  DIR="$(cd "$(dirname "$0")" >/dev/null 2>&1 && pwd)"
 
-OS=$(uname -s)
+  : "${ARCH:=$(uname -m)}"
+  : "${BRANCH:=master}"
+  : "${CTL_NAME:=kumactl}"
+  : "${PRODUCT_NAME:=Kuma}"
+  : "${REPO:=kumahq/kuma}"
+  : "${VERSION:=latest}"
+  : "${OS:=$(uname -s | tr '[:upper:]' '[:lower:]')}"
 
-if [ -z "$DISTRO" ]; then
-  DISTRO=""
-  if [ "$OS" = "Linux" ]; then
-    DISTRO=$(grep -oP '(?<=^ID=).+' /etc/os-release | tr -d '"')
-    if [ "$DISTRO" = "amzn" ]; then
-      DISTRO="centos"
+  OS="${OS:-linux}"
+  DISTRO="${DISTRO:-$OS}"
+
+  log "Welcome to the ${PRODUCT_NAME} automated download!"
+
+  case "_${OS}" in
+  _darwin) true ;;
+  _linux)
+    DISTRO="${DISTRO:-$(
+      grep -E '^ID=.*' /etc/os-release |
+        cut -d'=' -f2 |
+        tr -d '"' |
+        tr -d \' |
+        tr '[:upper:]' '[:lower:]'
+    )}"
+
+    if [ "$DISTRO" = 'amzn' ]; then
+      DISTRO='centos'
     fi
-  elif [ "$OS" = "Darwin" ]; then
-    DISTRO="darwin"
-  else
-    printf "ERROR\tOperating system %s not supported by %s\n" "$OS" "$PRODUCT_NAME"
-    exit 1
-  fi
-fi
+    ;;
+  _ | _*) err "Operating system ${OS} not supported by ${PRODUCT_NAME}." ;;
+  esac
 
-if [ -z "$DISTRO" ]; then
-  printf "ERROR\tUnable to detect the operating system\n"
-  exit 1
-fi
+  case "_${ARCH}" in
+  _amd64 | _x86_64)
+    ARCH='amd64'
+    ;;
+  _aarch64* | _arm64 | _armv8*)
+    ARCH='arm64'
+    ;;
+  _ | _*) err "Architecture ${ARCH} not supported by ${PRODUCT_NAME}" ;;
+  esac
 
-DETECTED_ARCH=$(uname -m)
-if [ "$ARCH" = "" ]; then
-  if [ "$DETECTED_ARCH" = "x86_64" ]; then
-    ARCH=amd64
-  elif [ "$DETECTED_ARCH" = "arm64" ] || \
-    [ "$DETECTED_ARCH" = "aarch64" ] || \
-    [ "$DETECTED_ARCH" = "aarch64_be" ] || \
-    [ "$DETECTED_ARCH" = "armv8l" ] || \
-    [ "$DETECTED_ARCH" = "armv8b" ]; then
-    ARCH=arm64
-  else
-    printf "ERROR\tArchitecture %s not supported by %s\n" "$DETECTED_ARCH" "$PRODUCT_NAME"
-    exit 1
-  fi
-fi
+  REPO_ORG="${REPO%%/*}"
+  REPO_REPO="${REPO##*/}"
 
-if [ -z "$VERSION" ]; then
-  # Fetching latest version
-  printf "INFO\tFetching latest %s version..\n" "$PRODUCT_NAME"
+  missing=''
+  for tool in curl grep gzip tar; do
+    if ! command -v $tool >/dev/null 2>&1; then
+      missing="${missing:+${missing} }${tool}"
+    fi
+  done
 
-  if ! VERSION=$(curl -s $LATEST_VERSION); then
-    printf "ERROR\tUnable to fetch latest %s version.\n" "$PRODUCT_NAME"
-    exit 1
+  if [ -n "$missing" ]; then
+    err "Required tools cannot be found: ${missing}."
   fi
 
-  if [ -z "$VERSION" ]; then
-    printf "ERROR\tUnable to fetch latest %s version because of a problem with %s.\n" "$PRODUCT_NAME" "$PRODUCT_NAME"
-    exit 1
+  if [ "$VERSION" = 'latest' ]; then
+    log "Fetching latest ${PRODUCT_NAME} version.."
+
+    LATEST_VERSION="$(curl -f -sL 'https://kuma.io/latest_version')"
```

Comment:
How do you expect this to work for KongMesh?

Reply:
I think you dropped the LATEST_VERSION url that we can change in the case of entreprise see: https://github.com/Kong/docs.konghq.com/blob/f31e8418de8657fd13182be828e5caab0ca81fcf/app/mesh/installer.sh#L24

Reply:
(worth actually checking this invocation locally :)

Reply:
Ah, you're right. I didn't realize it was being passed into the script externally like this.
I assumed I could just set `VERSION` to the correct version if unsupplied.
Good catch, I'll toss in a fix for this.